### PR TITLE
Rejecting the Promise when API calls return an error.

### LINKED ISSUE #6: improper handling of errors
For some reason Bitstamps returns the error below:
```js
{
  "status": 403,
  "headers": { },
  "body": {
    "status": "error",
    "reason": "Invalid signature",
    "code": "API0005"
  }
}
```
It is expected to have the result returned as an error and be passed to `catch`, however it is resolved as successful and passed to `then`

### PR DESCRIPTION
**Pull request by nicolasgarnier**
fixes #6

### CODE REVIEW

**Comment by tuloski** [comment]
@nicolasgarnier Be careful that not only body.status == "error" means error. Once I got the body only with "error" field (when the API keys doesn't have the permission for that action for example), and probably there are other cases like that. Probably it's an error on bitstamp side.

**Comment by nicolasgarnier** [comment]
Good feedback. I can add that check just in case although I think in that particular case the HTTP status should be 401 so we'll throw at [l.102](https://github.com/krystianity/node-bitstamp/pull/15/files#diff-a28a04559b68ecdd0d8184909711e86aR102)

**Comment by johnthillaye** [comment]
Is this getting merged ?

**Comment by tuloski** [comment]
Is this tested? Can I pull?

File: lib/Bitstamp.js [hunk 1]
```diff
@@ -90,15 +90,35 @@ class Bitstamp {
...
+                if (response.statusCode < 200 || response.statusCode > 299)
+                    return reject(new Error(body));
+                }
 
                 debug("result", options.url, response.statusCode);
                 try {
                     body = JSON.parse(body);
                 } catch(error){
-                    //empty
+                    reject(error);
```
Comment (krystianity) [inline lib/Bitstamp.js]: this is missing "return"

**Comment by krystianity** [comment]
Sry for the long wait guys, I reviewed and will merge asap if possible.

**Comment by nicolasgarnier** [comment]
done!

**Pull request merged**